unified-runtime/source/adapters/offload/enqueue.cpp

```diff
@@ -93,16 +93,19 @@ ur_result_t doWait(ur_queue_handle_t hQueue, uint32_t numEventsInWaitList,
   OL_RETURN_ON_ERR(makeEvent(TYPE, TargetQueue, hQueue, phEvent));
 
   if constexpr (Barrier) {
-    ol_event_handle_t BarrierEvent;
+    ur_event_handle_t BarrierEvent;
     if (phEvent) {
-      BarrierEvent = (*phEvent)->OffloadEvent;
+      BarrierEvent = *phEvent;
+      urEventRetain(BarrierEvent);
     } else {
-      OL_RETURN_ON_ERR(olCreateEvent(TargetQueue, &BarrierEvent));
+      OL_RETURN_ON_ERR(makeEvent(TYPE, TargetQueue, hQueue, &BarrierEvent));
```

Comment:
Nitpick: I know it's not added in this PR but the all caps name `TYPE` for a constexpr enum looks odd

Reply:
It's a template parameter, so in my head it inherits the naming scheme of enum variants which are all caps. Although I can't exactly say why I think it should work that way.